### ISSUE TITLE
CORS issues with Cloudflare-cached CSS

### ISSUE THREAD

**Issue by adelairdragon** [gryphonfluff/furtrack-tag-icons#1]
According to [Jack Raccoon](https://fursuits.online/@JackRacc@furry.engineer), who doesn't use GitHub, there is an issue where

> Sometimes CSS loads from Cloudflare and the domain is different enough it forbids cross-origin stuff

> Guess the Furtrack Tag Icons script will need to be updated to allow cross-origin permissions. It currently breaks with the new fontawesome version applied

There's a third potential clue according to him, where

> there's a fontawesome version mismatch. Cloudflare serves version 6.5.2 while Furtrack serves version 5.15.4

(All of this from the FurTrack discord).

These issues have supposedly started popping up since Saturday 2024-12-07's site update, which added BlueSky and Mastodon icons.

FWIW I have currently been unsuccessful in replicating but will advise if I can get a repro

**Comment by adelairdragon** [comment]
Found a repro (thanks Jack Raccoon):

* Install latest Firefox (currently 133.0), with ViolentMonkey (current version 2.29.0)
* Install the userscript
* Load the site; refresh if necessary
* Notice the following error:

```
Uncaught (in promise) DOMException: CSSStyleSheet.insertRule: Not allowed to access cross-origin stylesheet
    VMyy9cfbbrke moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:203
    fulfilled moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:182
    promise callback*step moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:184
    __awaiter moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:185
    __awaiter moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:181
    VMyy9cfbbrke moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:193
    VMyy9cfbbrke moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:278
    VMyy9cfbbrke moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:281
    VMyy9cfbbrke moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:283
    VMyy9cfbbrke moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:285
    rn moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/sandbox/injected-web.js:1
    <anonymous> moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/ FurTrack Tag Icons.user.js#1:1
    Le moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/sandbox/injected-web.js:1
    post moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/sandbox/injected-web.js:1
    ScriptData moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/sandbox/injected-web.js:1
    onHandle moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/sandbox/injected-web.js:1
    je moz-extension://1ada2ffa-de64-4f02-84a4-a558a4adcfce/sandbox/injected-web.js:1
FurTrack Tag Icons.user.js:203
```

**Comment by adelairdragon** [comment]
And this doesn't seem to be an issue with Chrome + TamperMonkey

**Comment by gryphonfluff** [comment]
This should be fixed in v0.2.1.

**Comment by adelairdragon** [comment]
Yep, Jack's confirming that it works now
